File: 048-self-powers.swift
import Foundation


let start = DispatchTime.now().uptimeNanoseconds

var sum = 0
for n in 1...1000 {
  var product = 1
  for _ in 0..<n {
    product *= n
    product %= 10_000_000_000
  }
  sum += product
  sum %= 10_000_000_000
}
print(sum)

let end = DispatchTime.now().uptimeNanoseconds
let seconds = Double(end - start) / 1_000_000_000
print("Took \(seconds) seconds")
// 0.00705290 seconds

File: 048-self-powers.c
#include <stdio.h>
#include <time.h>


int main(void)
{
	clock_t start = clock();

	unsigned long sum = 0;
	for (int n = 1; n <= 1000; ++n) {
		unsigned long product = 1;
		for (int i = 0; i < n; ++i) {
			product *= n;
			product %= 10000000000;
		}
		sum += product;
		sum %= 10000000000;
	}
	printf("%lu\n", sum);

	clock_t end = clock();
	float seconds = (float)(end - start) / CLOCKS_PER_SEC;
	printf("Took %f seconds\n", seconds);
	/** 0.006683 seconds */

	return 0;
}
